### ISSUE TITLE
Tooltip: research implementation

### ISSUE THREAD

**Issue by sstrubberg** [carbon-design-system/carbon#13934]
- DefinitionTooltip should be appear by default on hover and focus. Currently the prop has to be set to enable this.

Ref issue - https://github.com/carbon-design-system/carbon-website/issues/3332

> This is an omnibus ticket, summarizing key problems with the content of the Usage and Style pages for the Tooltip component. (Specific problems with tooltip _implementation_ are captured in https://github.com/carbon-design-system/carbon/issues/12921.)
 Detailed considerations, edits and suggestions are captured in pdf exports of the pages: [usage.pdf ](https://ibm.box.com/s/p07zoqxau8xw7ounitiwr9ha2i5tyqsj)and [style.pdf](https://ibm.box.com/s/6utyxjt68q1757mn9twx01rn1s97wekt). These documents should be reviewed in detail.

> The key considerations are:

> - It is recommended that Carbon make the definition tooltip always display on hover and focus. You _could_ consider making it also display on click/activation (since AFAIK there is not other action associated with a defn tooltip) but it would be less likely to muff people up if it was just hover/focus. Let's make tooltip consistent for interaction, regardless of the context.
> - This is a lesser priority, but Carbon could consider providing an example/implementation of a tooltip on an input. [Option 4 in this demonstration](https://a11y-style-guide.com/style-guide/section-forms.html#kssref-forms-tooltips) provides a poor-ish example. The potential advantage of this is where you don't want the user to have to activate an inforamtion icon to toggle on more pronounced help. 👈 not a priority

```[tasklist]
- [ ] Update storybook to have `openOnHover` true for both stories
```
